functional-test/src/test/groovy/org/rundeck/tests/functional/util/WaitUtilsSpec.groovy

```diff
@@ -0,0 +1,88 @@
+package org.rundeck.tests.functional.util
+
+import org.rundeck.util.annotations.APITest
+import org.rundeck.util.common.ResourceAcceptanceTimeoutException
+import org.rundeck.util.common.WaitUtils
+import spock.lang.Specification
+
+@APITest
+class WaitUtilsSpec extends Specification {
+
+    static final Map<String, String> TEST_VALUES = ["1": "one", "2": "two"].asImmutable()
+
+    def testWaitForThatNeverWaits() {
+        given:
+        Closure<String> retriever = { return TEST_VALUES[it] }
+
+        when:
+        String result = WaitUtils.waitForResource("1", retriever)
+
+        then:
+        result == "one"
```

Comment:
How do we make sure that the code (or future changes) never actually waited?

Reply:
Changed to wait for `0`.